### PR TITLE
Review code

### PR DESCRIPTION
**Pull request by sareepattamannil**

### CODE REVIEW

File: Calculator.html [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+    <script src="Calculator.js"></script>
+    <div>
+        <div class="screen" id="screen">Screen</div>
+        <div class="btn">
+            <button id="1" onclick="number('1')">1</button>
+            <button id="2" onclick="number('2')">2</button>
+            <button id="3" onclick="number('3')">3</button>
+            <button id="+" onclick="operator1('+')">+</button>
 
-            <button style="width:25%" id="4" onclick="number('4')">4</button>
```
Comment (sareepattamannil) [inline Calculator.html:15]: Moved styles to the css file

File: Calculator.css [hunk 2]
```diff
@@ -10,13 +10,27 @@ body {
...
+.container{
+    width: 150px; /* Fixed width for the grid */
+}
+
+.btn{
+    display: grid;
+    grid-template-columns: repeat(4, 1fr); /* Four equal columns */
+    gap: 10px; /* Space between grid items */
+}
+
```
Comment (sareepattamannil) [inline Calculator.css:22]: Used grid to create 4 column style

File: Calculator.css [hunk 3]
```diff
@@ -10,13 +10,27 @@ body {
     padding: 50px;
 }
 
+.container{
+    width: 150px; /* Fixed width for the grid */
+}
```
Comment (sareepattamannil) [inline Calculator.css:15]: Wrapped the whole calculator to give it a fixed width

File: Calculator.js [hunk 4]
```diff
@@ -5,71 +5,60 @@ let op1;
...
+  op1 = "";
+  done = false;
+}
 
 function number(num) {
+  if (done) cleared();
 
-    if(done === true) {
-        cleared()
-    }
```
Comment (sareepattamannil) [inline Calculator.js:21]: Anything that evaluates to a `truthy` value can be checked without the fully condition of writing `done === true`

File: Calculator.js [hunk 5]
```diff
@@ -5,71 +5,60 @@ let op1;
...
-        cleared();
-    };
-    if (document.getElementById("screen").innerHTML.includes(".")) {
-        document.getElementById("screen").innerHTML = "ERROR!"
-    };
-
-    if(document.getElementById("screen").innerHTML === "") {
-        document.getElementById("screen").innerHTML = "0.";
-    } else {
-        number('.');
```
Comment (sareepattamannil) [inline Calculator.js:56]: Can simplify this logic by pulling out:
```js
  const screen = document.getElementById("screen");
  ```

File: Calculator.js [hunk 6]
```diff
@@ -5,71 +5,60 @@ let op1;
...
-function enter() {
-
-    if (oper === true) {
-        if (op1 === "^") {
-            var2 = document.getElementById("screen").innerHTML;
-            result = var1 ** var2;
-        } else {
-            var2 = document.getElementById("screen").innerHTML; 
-            result = eval(+var1 + op1 + +var2);
-        };
```
Comment (sareepattamannil) [inline Calculator.js:69]: Eval is generally a security issue but in this case doesn't matter, can always consider a switch statement:

```js
switch (operator) {
    case "+":
        result = +firstOperand + +secondOperand;
        break;
    case "-":
        result = +firstOperand - +secondOperand;
        break;
    case "*":
        result = +firstOperand * +secondOperand;
        break;
    case "/":
        if (+secondOperand === 0) {
            result = "ERROR!"; // Handle division by zero
        } else {
            result = +firstOperand / +secondOperand;
        }
        break;
    case "^":
        result = Math.pow(+firstOperand, +secondOperand);
        break;
    default:
        result = "ERROR!"; // Handle unknown operators
        break;
}

document.getElementById("screen").innerHTML = result;
calculationDone = true;
```

**Review by LoneLombax1 — APPROVED** [review]

**Pull request merged**